pkg/api/types.go

```diff
@@ -374,6 +374,18 @@ func (a *AgentPoolProfile) IsLinux() bool {
 	return a.OSType == Linux
 }
 
+// IsDistroCoreOS returns true if distro is coreos
+func (a *AgentPoolProfile) IsDistroCoreOS() bool {
+	return strings.EqualFold(a.Distro, "coreos")
+}
+
+// IsDistroUbuntu returns true if distro is ubuntu or not specified.
+//IsDistroUbuntu returns treue if Distro in the agentppol is Ubuntu or Not spefcified
+func (a *AgentPoolProfile) IsDistroUbuntu() bool {
+
+	return (strings.EqualFold(a.Distro, "ubuntu") || a.Distro == "")
```

Comment:
What happens in case of OsType == Windows? In that case IsDistroUbuntu should return false.